settings.gradle

```diff
@@ -270,6 +270,9 @@ project(':extensions-s3').projectDir = file('extensions/s3')
 include(':extensions-iceberg')
 project(':extensions-iceberg').projectDir = file('extensions/iceberg')
 
+include(':extensions-iceberg-aws')
+project(':extensions-iceberg-aws').projectDir = file('extensions/iceberg-aws')
```

Comment:
Should it be `iceberg/aws`?

Reply:
nest subdir vs nested subproject - technically orthogonal. ryan's comment is asking for the former, which is a way to structure the directories (replace the string passed to the file() method). I would argue that not only do we want that, we _also_ want nested subprojects, using `:` delimiters.

The Gradle convention is that nested directories also always means nested projects. Deephaven (starting with DHE, inherited in DHC) doesn't follow this as a rule, though DHC _sometimes_ uses this _if_ the parent directory is itself a project.

That said, using subprojects also conflicts a bit with our naming policies (which is "deephaven-" plus the project's own name), since nesting projects removes the parent project from the name.

(for the rest of this i'm assuming we're calling it `s3` rather than `aws`, but `aws-s3` might also make sense)

From talking to Larry it sounds like we anticipate other iceberg implementations that aren't s3 specific. So, I would propose something like:
```
extensions/
  iceberg/ # name this project deephaven-extensions-iceberg?
    build.gradle
    src/
      s3/ # name this project deephaven-iceberg-s3? 
         build.gradle
         src/
      r2/ # possible other impl?
        build.gradle
        src/